sky/packages/sky/lib/src/animation/animated_value.dart

```diff
@@ -15,7 +15,10 @@ enum Direction {
   reverse
 }
 
-/// A variable that changes as an animation progresses
+/// An interface describing a variable that changes as an animation progresses.
+/// 
+/// AnimatedVariables, by convention, must be cheap to create. This allows them to be used in 
+/// build functions in Widgets.
```

Comment:
If the widget that contains the transition is itself in something that is animated, it'll be built every frame.

A good rule of thumb is to assume build() and syncConstructionArguments() are called every frame.